junit-jupiter-api/src/main/java/org/junit/jupiter/api/Assertions.java

```diff
@@ -135,6 +135,9 @@ public static <V> V fail() {
 	 * }</pre>
 	 */
 	public static <V> V fail(String message) {
+		if (message.isBlank()) {
```

Comment:
`AssertionUtils.fail` is fine accepting `null`, but here `fail` will fail with NPEx for `null` message.

Similar below.